scripts/create_sfm_from_obj.py

```diff
@@ -0,0 +1,145 @@
+        R = Rs[i].unsqueeze(0)
+        T = Ts[i].unsqueeze(0)
+        image = render_image(mesh, fx, fy, cx, cy, W, H, R, T)
+        image = (image[..., :3] * 255.).astype(np.uint8)
+        image = cv2.cvtColor(image, cv2.COLOR_RGB2BGR)
+        img_name = ('%d.png' % (i + 1)).zfill(8)
+        cv2.imwrite(str(mapping_path / img_name), image)
+
+        colmap_image_name = 'mapping/' + img_name
+        colmap_image_id = i + 1
+        colmap_image = create_colmap_image_from_pytorch3d_RT(R[0], T[0], 
+                                                          colmap_image_name,
+                                                          colmap_image_id,
+                                                          colmap_camera_id)
+        colmap_images[colmap_image_id] = colmap_image
+    return colmap_images, colmap_cameras
+
+def create_sfm_from_colmap_images_and_cameras(dataset_path, output_path,
+                                              colmap_images, colmap_cameras): 
+    # Configure SFM
+    images = Path(dataset_path)
+    outputs = Path(output_path)
+    sfm_pairs = outputs / 'pairs-sfm.txt'
+    loc_pairs = outputs / 'pairs-loc.txt'
+    features = outputs / 'features.h5'
+    matches = outputs / 'matches.h5'
+    raw_dir = outputs / "raw"
+    sfm_dir = outputs / "sfm"
+    feature_conf = extract_features.confs['superpoint_max']
+    matcher_conf = match_features.confs['superglue']
+    matcher_conf['model']['weights'] = 'indoor'
+    references = [str(p.relative_to(images)) for p in (images / 'mapping/').iterdir()]
+    print(len(references), "mapping images")
+    
+    # Extract features and matches
+    extract_features.main(feature_conf, images, image_list=references, feature_path=features)
+    pairs_from_exhaustive.main(sfm_pairs, image_list=references)
+    match_features.main(matcher_conf, sfm_pairs, features=features, matches=matches)
+    pairs = sfm_pairs
+    assert features.exists(), features
+    assert pairs.exists(), pairs
+    assert matches.exists(), matches
+
+    # Create a database and import images, features and matches
+    raw_dir.mkdir(parents=True, exist_ok=True)
+    database = raw_dir / 'database.db'
+    create_empty_db(database)
+    camera_mode = 'AUTO'
+    import_images(image_dir=images, 
+                  database_path=database, 
+                  camera_mode=camera_mode, 
+                  image_list=None)
+    image_ids = get_image_ids(database)
+    import_features(image_ids, database, features)
+    import_matches(image_ids, database, pairs, matches,
+                   min_match_score=None, 
+                   skip_geometric_verification=False)
+    geometric_verification(database, pairs, verbose=True)
+
+    # Create images, cameras and points3D binaries
+    write_images_binary(colmap_images, raw_dir / 'images.bin')
+    write_cameras_binary(colmap_cameras, raw_dir / 'cameras.bin')
+    write_points3D_binary({}, raw_dir / 'points3D.bin') #Should be empty
```

Comment:
How hard would it be to support this? Given I might need 3d points for the depth stuff. 
triangulation add that info to the dir?

Reply:
Triangulation (the next step) fills `points3D.bin` with 3d points. The SfM created is identical to the real SfMs we've created as part of our initial experiments.